src/main/app/controller/BasePaymentPostController.ts

```diff
@@ -105,14 +105,26 @@ export default abstract class BasePaymentPostController {
       CITIZEN_ADD_PAYMENT
     );
 
-    return payment;
+    return redirectUrl;
+  }
+
+  public async findServiceRequestReference(feeCode: string, client: PaymentClient): Promise<string | undefined> {
```

Comment:
I don't understand this bit.
Is it calling into f&p and only find the service request made for this case? Or is it finding a service request made with the fee code?

Reply:
thinking we need to be careful we don't just grab a service request but that it belongs to someone else's case

Reply:
Hi @DelythJustice, we get the payment groups for their case by calling `/cases/${caseId}/paymentgroups` in [PaymentClient.getCasePaymentGroups()](https://github.com/hmcts/nfdiv-frontend/pull/3860/files#diff-78fbef39087b4e70a8b5cb5892d7f1b0fdd5776fbbd568dbc5b746a3a422c384R31). CaseId hasn't been passed as an arg here because we can get it from the user session in payment client

Reply:
ta